src/main/java/impl/image/ImageConverterImpl.java

```diff
@@ -0,0 +1,171 @@
+     * The fallback just does what's obvious approach - sets various bit parts of a number.
+     */
+    private Function<Color, Integer> getColorValueGetterFunction() {
+        if (colorValueGetter == null) {
+            try {
+                var field = Color.class.getDeclaredField("value");
+
+                if (valueIsPrivate) {
+//                  Yeah, the fun part about java is that if you get the field, `setAccessible` will never throw
+                    field.setAccessible(true);
+                    valueIsPrivate = false;
+                }
+
+                colorValueGetter = (Color color) -> {
+                    try {
+                        return (int) field.get(color);
+                    } catch (IllegalAccessException e) {
+                        e.printStackTrace();
+//            Weird fallback, for some reason there's no way here to use it as a functor and apply like w'd really want.
+                        return deconstructColor(color);
+                    }
+                };
+            } catch (NoSuchFieldException e) {
+                colorValueGetter = this::deconstructColor;
+            }
+        }
+        return colorValueGetter;
+    }
+
+    /**
+     * Works sorta like a one-time generated property with backing field
+     * Gets a function that sets value field of a Color object.
+     * If reflection access was OK - it's direct setting to the private field of a freshly created Color object
+     * The fallback just does what's obvious approach - gets different bit parts and constructs proper object
+     */
+    private Function<Integer, Color> getColorValueSetterFunction() {
+        if (colorValueSetter == null) {
+
+            try {
+                var field = Color.class.getDeclaredField("value");
+
+                if (valueIsPrivate) {
+//                  Yeah, the fun part about java is that if you get the field, `setAccessible` will never throw
+                    field.setAccessible(true);
+                    valueIsPrivate = false;
+                }
+
+                colorValueSetter = (Integer value) -> {
+                    try {
+                        var color = new Color(0);
+                        field.set(color, value);
+                        return color;
+                    } catch (IllegalAccessException e) {
+                        e.printStackTrace();
+//            Weird fallback, for some reason there's no way here to use it as a functor and apply like w'd really want.
+                        return constructColor(value);
+                    }
+                };
+            } catch (NoSuchFieldException e) {
+                colorValueSetter = this::constructColor;
+            }
+        }
+        return colorValueSetter;
+    }
+
+
+    /**
+     * Transforms 2d array of type S to 2d array of type T
+     * @param <S> Source type
+     * @param <T> Target type
+     */
+    private static class ArrayTransformer<S, T> {
```

Comment:
There is the table which describes the order that class parts should appear: https://www.oracle.com/technetwork/java/javase/documentation/codeconventions-141855.html#1852
There is no information about nested classes, but they should be after other parts

Reply:
Hmm, ok